File: 0067-add-binary/0067-add-binary.py
class Solution:
    def addBinary(self, a: str, b: str) -> str:
        res = ""
        i = len(a) - 1
        j = len(b) - 1
        carry = 0
        while i >= 0 or j >= 0:
            x = 0
            y = 0
            if i >= 0:
                x = int(a[i])
                i -= 1
            if j >= 0:
                y = int(b[j])
                j -= 1
            res = str(x ^ y ^ carry) + res
            carry = (x + y + carry) // 2
        if carry == 1:
            res = '1' + res
        return res

File: 0067-add-binary/0067-add-binary.cpp
class Solution {
public:
    string addBinary(string a, string b) {
        string res = "";
        int i = a.length() - 1;
        int j = b.length() - 1;
        char carry = 0;
        while(i >= 0 || j >= 0)
        {
            char x = 0;
            char y = 0;
            if(i >= 0)
            {
                x = a[i] - '0';
                i--;
            }
            if(j >= 0)
            {
                y = b[j] - '0';
                j--;
            }
            res = to_string(x ^ y ^ carry) + res;
            carry = (x + y + carry) / 2;
        }
        if(carry == 1)
        {
            res = "1" + res;
        }
        return res;
    }
};